src/nupic/utils/StringUtils.cpp

```diff
@@ -394,17 +394,17 @@ bool StringUtils::toIntListNoThrow(const std::string &s, std::vector<Int> &list,
 
 
 //--------------------------------------------------------------------------------
-std::shared_ptr<Byte> StringUtils::toByteArray(const std::string &s,  Size bitCount) {
+std::shared_ptr<char> StringUtils::toByteArray(const std::string &s,  Size bitCount) {
```

Comment:
not in this PR, but could we change these to `string` return types? Even if that breaks compat, I think it would be simpler code. @dkeeney?

Reply:
Ok, I just now reverted this change.  The reason I replaced Byte with char was to make pybind11 happy with the type conversions, which I have since fixed in a different way.